main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateRouteNameValidator.java

```diff
@@ -0,0 +1,88 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.DuplicateRouteNameNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsRoute;
+import org.mobilitydata.gtfsvalidator.table.GtfsRouteTableContainer;
+
+/**
+ * Validates unicity of short and long name for all routes.
+ *
+ * <p>Generated notice:
+ *
+ * <ul>
+ *   <li>{@link DuplicateRouteNameNotice}
+ * </ul>
+ */
+@GtfsValidator
+public class DuplicateRouteNameValidator extends FileValidator {
+  @Inject GtfsRouteTableContainer routeTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    final Map<String, GtfsRoute> routeByRouteLongName = new HashMap<>(routeTable.entityCount());
+    final Map<String, GtfsRoute> routeByShortName = new HashMap<>(routeTable.entityCount());
+    routeTable
+        .getEntities()
+        .forEach(
+            route -> {
+              if (route.hasRouteLongName()) {
+                if (routeByRouteLongName.containsKey(route.routeLongName())) {
+                  if (areRoutesFromSameAgency(
+                      route.agencyId(),
+                      routeByRouteLongName.get(route.routeLongName()).agencyId())) {
+                    noticeContainer.addValidationNotice(
+                        new DuplicateRouteNameNotice(
+                            "route_long_name", route.csvRowNumber(), route.routeId()));
+                  }
+                } else {
+                  routeByRouteLongName.put(route.routeLongName(), route);
+                }
+              }
+              if (route.hasRouteShortName()) {
+                if (routeByShortName.containsKey(route.routeShortName())) {
+                  if (areRoutesFromSameAgency(
+                      route.agencyId(), routeByShortName.get(route.routeShortName()).agencyId())) {
+                    noticeContainer.addValidationNotice(
+                        new DuplicateRouteNameNotice(
+                            "route_short_name", route.csvRowNumber(), route.routeId()));
+                  }
+                } else {
+                  routeByShortName.put(route.routeShortName(), route);
+                }
+              }
+            });
+  }
+
+  /**
+   * Determines if two routes are from the same agency
+   *
+   * @param routeAgencyId first agency_id
+   * @param otherRouteAgencyId second agency_id
+   * @return true if both agency ids are equals returns false otherwise.
+   */
+  private boolean areRoutesFromSameAgency(
+      final String routeAgencyId, final String otherRouteAgencyId) {
+    return routeAgencyId.equalsIgnoreCase(otherRouteAgencyId);
```

Comment:
IDs are case-sensitive. Also, it may be useful to show warnings if several agencies have the same ID ignoring case.

Reply:
I remember debating if we should treat IDs as case-sensitive. So I assume Google treats them as case-sensitive?

In the spec, the only mention I could find is:

>All field names are case-sensitive.

...but it doesn't say anything about field values.

@MobilityData/transit-specs another issue that needs clarification in the spec.

Reply:
The spec says: "An ID field value [...] is a sequence of any UTF-8 characters." ALL-CAP letters and small-cap letters are different UTF-8 characters. So IMHO, ID values are implicitly case sensitive. Is there an ERROR or a WARNING issued in the validator on this matter?

Registered in Discrepancies Specification vs Validator MobilityData/gtfs-tasks#87

Reply:
@timMillet as discussed, at present the validator does not issue any notice if two agencies have the same value for `agency.agency_id`. As you mentioned, `route id VALUE` and `route ID vaLUe` are two different strings and should not be considered as equal. I'm opening an issue on this matter to improve de validator's behavior.